packages/lodestar/src/sync/utils/sync.ts

```diff
@@ -218,13 +218,46 @@ export async function syncPeersStatus(reps: IReputationStore, network: INetwork,
   }));
 }
 
-export function getBestHead(peers: PeerId[], reps: IReputationStore): {slot: number; root: Root} {
+/**
+ * Check if a peer supports sync.
+ */
+export async function checkPeerSupportSync(
+  config: IBeaconConfig, reps: IReputationStore, peerId: PeerId, reqResp: IReqResp): Promise<void> {
+  const latestStatus = reps.getFromPeerId(peerId).latestStatus;
+  if (!latestStatus) {
+    reps.getFromPeerId(peerId).supportSync = false;
+    return;
+  }
+  const finalizedSlot = computeStartSlotAtEpoch(config, latestStatus.finalizedEpoch);
+  const testCount = 5;
+  if (latestStatus.headSlot - finalizedSlot <= testCount) {
+    reps.getFromPeerId(peerId).supportSync = false;
+    return;
+  }
+  const testReqResp: BeaconBlocksByRangeRequest = {
+    startSlot: finalizedSlot,
+    count: testCount,
+    step: 1
+  };
+  const result = await reqResp.beaconBlocksByRange(peerId, testReqResp);
+  // a lot of good peers return testCount - 1 = 4 maybe because of missing slot
```

Comment:
This is quite tricky check, we could have hundreds of skipped slots which would stall us. I think we should just check of unsupported protocol here and in sync store peers which returned 0 block chunks and if we receive blocks after that than remove blocks by range from supported protocols

Reply:
yeah so I changed it to:
+ get finalized block
+ get finalized block's parent
+ `beacon_blocks_by_range` from parent to finalized block, confirm it returns 2

this should work in all scenarios, works fine in my test